homeassistant/components/switchbot/__init__.py

```diff
@@ -29,57 +37,67 @@
+    ATTR_CURTAIN: switchbot.SwitchbotCurtain,
+    ATTR_BOT: switchbot.Switchbot,
+}
+
+_LOGGER = logging.getLogger(__name__)
 
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Switchbot from a config entry."""
     hass.data.setdefault(DOMAIN, {})
+    domain_data = hass.data[DOMAIN]
+
+    if CONF_ADDRESS not in entry.data and CONF_MAC in entry.data:
+        # Bleak uses addresses not mac addresses which are are actually
+        # UUIDs on some platforms (MacOS).
+        mac = entry.data[CONF_MAC]
+        if "-" not in mac:
+            mac = dr.format_mac(mac)
+        hass.config_entries.async_update_entry(
+            entry,
+            data={**entry.data, CONF_ADDRESS: mac},
+        )
 
     if not entry.options:
-        options = {
-            CONF_TIME_BETWEEN_UPDATE_COMMAND: DEFAULT_TIME_BETWEEN_UPDATE_COMMAND,
-            CONF_RETRY_COUNT: DEFAULT_RETRY_COUNT,
-            CONF_RETRY_TIMEOUT: DEFAULT_RETRY_TIMEOUT,
-            CONF_SCAN_TIMEOUT: DEFAULT_SCAN_TIMEOUT,
-        }
-
-        hass.config_entries.async_update_entry(entry, options=options)
-
-    # Use same coordinator instance for all entities.
-    # Uses BTLE advertisement data, all Switchbot devices in range is stored here.
-    if DATA_COORDINATOR not in hass.data[DOMAIN]:
-
-        if COMMON_OPTIONS not in hass.data[DOMAIN]:
-            hass.data[DOMAIN][COMMON_OPTIONS] = {**entry.options}
-
-        switchbot.DEFAULT_RETRY_TIMEOUT = hass.data[DOMAIN][COMMON_OPTIONS][
-            CONF_RETRY_TIMEOUT
-        ]
-
-        # Store api in coordinator.
-        coordinator = SwitchbotDataUpdateCoordinator(
-            hass,
-            update_interval=hass.data[DOMAIN][COMMON_OPTIONS][
-                CONF_TIME_BETWEEN_UPDATE_COMMAND
-            ],
-            api=switchbot,
-            retry_count=hass.data[DOMAIN][COMMON_OPTIONS][CONF_RETRY_COUNT],
-            scan_timeout=hass.data[DOMAIN][COMMON_OPTIONS][CONF_SCAN_TIMEOUT],
+        hass.config_entries.async_update_entry(
+            entry,
+            options={
+                CONF_RETRY_COUNT: DEFAULT_RETRY_COUNT,
+                CONF_RETRY_TIMEOUT: DEFAULT_RETRY_TIMEOUT,
+            },
         )
 
-        hass.data[DOMAIN][DATA_COORDINATOR] = coordinator
-
-    else:
-        coordinator = hass.data[DOMAIN][DATA_COORDINATOR]
+    sensor_type: str = entry.data[CONF_SENSOR_TYPE]
+    address: str = entry.data[CONF_ADDRESS]
+    ble_device = bluetooth.async_ble_device_from_address(hass, address.upper())
+    if not ble_device:
+        raise ConfigEntryNotReady(
+            f"Could not find Switchbot {sensor_type} with address {address}"
+        )
 
-    await coordinator.async_config_entry_first_refresh()
+    if COMMON_OPTIONS not in domain_data:
+        domain_data[COMMON_OPTIONS] = entry.options
 
-    entry.async_on_unload(entry.add_update_listener(_async_update_listener))
+    common_options: dict[str, int] = domain_data[COMMON_OPTIONS]
```

Comment:
```suggestion
    common_options: Mapping[str, int] = domain_data[COMMON_OPTIONS]
```

Reply:
There is no point in having common options anymore since each one can have its own.  And that makes more sense since the battery powered devices likely might need different adjustments if they have range issues

Reply:
The retry timeout is likely because the connect timeout is too low which means [services cannot be resolved in time](https://github.com/hbldh/bleak/blob/0361ba4b7e4682f4e97ecacaab4499b84c70f288/bleak/backends/bluezdbus/client.py#L316) so it disconnects in the middle of the connection leaving things in a bad state

Reply:
I opened a pr with the lib to fix that https://github.com/Danielhiversen/pySwitchbot/pull/50